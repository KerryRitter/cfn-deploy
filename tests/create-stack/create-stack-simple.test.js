const mockCreateStack = require('../mocks/createStack');
const lib = require('../..');


describe('cfn-deploy', () => {
  beforeAll(() => mockCreateStack.beforeAll());

  it('should successfully create new stack', (done) => {
    const events = lib({
      region: 'us-east-1',
      stackName: 'new-stack',
      template: './tests/templates/simple-template.yaml',
    });
    events.on('ERROR', err => done(err));
    events.on('COMPLETE', (data) => {
      expect(data).toMatchSnapshot();
    });
    events.on('FINALLY', done);
  });

  afterAll(() => mockCreateStack.afterAll());
});
