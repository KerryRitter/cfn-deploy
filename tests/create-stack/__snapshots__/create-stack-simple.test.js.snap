// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cfn-deploy should successfully create new stack 1`] = `
Object {
  "Capabilities": undefined,
  "ChangeSetName": "cfn-deploy",
  "ChangeSetType": "CREATE",
  "Description": "Created with cfn-deploy",
  "Parameters": Array [],
  "StackName": "new-stack",
  "TemplateBody": "Resources:
  S3Bucket:
    Type: AWS::S3::Bucket
",
}
`;

exports[`cfn-deploy should successfully create new stack 2`] = `
Object {
  "type": "UPDATE",
}
`;
